sktime/transformations/series/bootstrapping.py

```diff
@@ -0,0 +1,395 @@
+# -*- coding: utf-8 -*-
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Bootstrapping methods for time series."""
+
+__author__ = ["ltsaprounis"]
+
+from copy import copy
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+from statsmodels.tsa.api import STL as _STL
+
+from sktime.transformations.base import BaseTransformer
+from sktime.transformations.series.boxcox import BoxCoxTransformer
+
+
+class UnivariateBootsrappingTransformer(BaseTransformer):
+    """Creates a population of similar time series.
+
+    This method utilises a form of bootstrapping to generate a population of
+    similar time series to the input time series [1]_, [2]_.
+
+    First the observed time series is transformed using a Box-Cox transformation to
+    stabilise the variance. Then it's decomposed to seasonal, trend and remainder
+    time series, using the STL algorithm from statsmodels [4]_. We then sample blocks
+    from the remainder time series using the Moving Block Bootstrapping (MBB) method
+    [3]_ to create synthetic remainder series that mimic the autocorrelation patterns
+    of the observed series. Finally these bootstrapped remainders are added to the
+    season and trend components and we use the inverse Box-Cox transform to return
+    a panel of similar time series.
+
+    Parameters
+    ----------
+    number_of_new_series : int, optional
+        The number of bootstraped time series that will be generated, by default 10
+    sp : int, optional
+        Seasonal periodicity of the data in integer form, by default 12.
+        Must be an integer >= 2
+    block_length : int, optional
+        The length of the block in the MBB method, by default None.
+        If not provided, the following heuristic is used, the block length will the
+        minimum between 2*sp and len(X) - sp.
+    sampling_replacement: bool, optional
+        Whether the MBB sample is with or without replacement, by default False.
+    return_actual : bool, optional
+        If True the output will contain the actual time series, by default True.
+        The actual time series will be labelled as "<series_name>_actual" (or "actual"
+        if series name is None).
+    series_name : str, optional
+        The series name, by default None
+        If provided, the synthetic series names will have the series_name as a
+        prefix followed by an undescore e.g. "<series_name>_synthetic_1". If not
+        provided the series names will no prefix will be present e.g. "synthetic_1".
+    boxcox_bounds : Tuple, optional
+        Lower and upper bounds used to restrict the feasible range
+        when solving for the value of lambda, by default None.
+    boxcox_method : str, optional
```

Comment:
```suggestion
    bounds : Tuple, optional
        Lower and upper bounds used to restrict the feasible range
        when solving for the value of lambda, by default None.
    method : str, optional
```
I would keep original bixcox param names and rather describe in docstring that it goes into boxcox

Reply:
changed it to `lambda_bounds` & `lambda_method`